docs/deployment/image.md

```diff
@@ -0,0 +1,18 @@
+# Build your instance Docker image
+
+In this section you will learn to build the docker image of your InvenioRDM instance! Doing so is as simple as running the following command:
+
+``` console
+docker build -t demo-inveniordm /path/to/your/instance --build-arg include_assets=true
+```
+
+!!! info "Tag name"
+    In this case we have chosen to call our image `demo-inveniordm`, but you can choose your own name.
+
+**Why is the `include_assets` flag needed**
+
+When running InvenioRDM in local (`invenio-cli run`) you are running the uWSGI server in your own machine. On the other hand, when running `invenio-cli containerize` you are building the docker image. Then you are running it in your machine along the other containers. Nonetheless, the statics are built in a different fashion. However, when deploying somewhere else they are needed.
+
+This is the case when deploying in OpenShift. To solve this problem the charts define a shared volume mounted on the Nginx and the Web containers.
+
+When a volume is mounted it overwrites the contents of the folder (i.e. it will delete the assets). Therefore, we make use of an `initContainer` to mount the volume in a temporary location and copy to the volume the assets. This way when the volume is mounted into Nginx and Web containers it already has the assets. Being the end result that `/static` files can be served directly by Nginx.
```

Comment:
@fenekku I think the key point is that the `invenio-assets` container corresponds to the [invenio-assets-claim](https://github.com/inveniosoftware/helm-invenio/blob/master/invenio/templates/volumes.yaml#L5) persistent volume. As I perceived it is:
1- spin up an initContainer
2- mount the invenio-assets persistent voume in a temporary location. This means the volume is empty initially and when it is mounted first it overrides whatever is in the location is mounted with its contents (which are empty in the beginning).
3 - copy static assets into the container. 
4- initiContainer is umounted along with the volume but because is a persistent volume the data are persisted
5 - Web container is mounting again the `invenio-assets` but now the volume overrides whatever is in the location it is mounted BUT it includes already the desired data.

@ppanero can confirm or tell me how wrongly I have understood what is going on :)

Reply:
@fenekku bad news first:

> Mounting in OpenShift deletes assets?

Yes... 😢 

The steps mentioned by Zach are correct. In step 5 also nginx mounts it, it is nginx who serves the statics in the end. The difference form building statics on real containers is as follows:

- Right now we build them in the image (time consumed once), then is "just" a `cp` operation when deploying and the shared volume has the assets. Therefore, al nginx/web nodes have them.

- If we build them in the real containers, means that each time a new pod/container is spawned it needs to collect, create, potentially install and build assets. Taking aside the fact that we would be "re-doing" the step 6 times (in the cloud) instead of in the PC (which has more resources than pods). The main issue here is that pods are usually spawned "ad-hoc" when you have a spike in usage (i.e. when you need more processing fast) and having to `buildall` at that point is counterproductive (consumes extra resources) and really slow.

Reply:
Thanks for the explainer! Got it now.